Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/HttpUtility.cs

```diff
@@ -0,0 +1,670 @@
+            int value = 0;
+            int end = length + offset;
+            for (int i = offset; i < end; i++) {
+                int current = GetInt (bytes [i]);
+                if (current == -1)
+                    return -1;
+                value = (value << 4) + current;
+            }
+
+            return value;
+        }
+
+        static int GetChar (string str, int offset, int length)
+        {
+            int val = 0;
+            int end = length + offset;
+            for (int i = offset; i < end; i++) {
+                char c = str [i];
+                if (c > 127)
+                    return -1;
+
+                int current = GetInt ((byte) c);
+                if (current == -1)
+                    return -1;
+                val = (val << 4) + current;
+            }
+
+            return val;
+        }
+
+        public static string UrlDecode (byte [] bytes, int offset, int count, Encoding e)
+        {
+            if (bytes == null)
+                return null;
+            if (count == 0)
+                return String.Empty;
+
+            if (bytes == null)
+                throw new ArgumentNullException ("bytes");
+
+            if (offset < 0 || offset > bytes.Length)
+                throw new ArgumentOutOfRangeException ("offset");
+
+            if (count < 0 || offset + count > bytes.Length)
+                throw new ArgumentOutOfRangeException ("count");
+
+            StringBuilder output = new StringBuilder ();
+            MemoryStream acc = new MemoryStream ();
+
+            int end = count + offset;
+            int xchar;
+            for (int i = offset; i < end; i++) {
+                if (bytes [i] == '%' && i + 2 < count && bytes [i + 1] != '%') {
+                    if (bytes [i + 1] == (byte) 'u' && i + 5 < end) {
+                        if (acc.Length > 0) {
+                            output.Append (GetChars (acc, e));
+                            acc.SetLength (0);
+                        }
+                        xchar = GetChar (bytes, i + 2, 4);
+                        if (xchar != -1) {
+                            output.Append ((char) xchar);
+                            i += 5;
+                            continue;
+                        }
+                    } else if ((xchar = GetChar (bytes, i + 1, 2)) != -1) {
+                        acc.WriteByte ((byte) xchar);
+                        i += 2;
+                        continue;
+                    }
+                }
+
+                if (acc.Length > 0) {
+                    output.Append (GetChars (acc, e));
+                    acc.SetLength (0);
+                }
+
+                if (bytes [i] == '+') {
+                    output.Append (' ');
+                } else {
+                    output.Append ((char) bytes [i]);
+                }
+            }
+
+            if (acc.Length > 0) {
+                output.Append (GetChars (acc, e));
+            }
+
+            acc = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'acc'.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)